### PR TITLE
Upgrade setCameraOrientation to setCameraPose

### PR DESCRIPTION
**Pull request by rajat2004**
Replaces #2624 

From the original PR - 

setCameraPose now takes a pose argument instead of just orientation. Commanded pose is expected to contain position in NED, **relative to default camera location.**

Status:

- [x] Base implementation of setCameraPose for Unreal
- [x] Testing in Unreal
- [x] Unity implementation
- [x] Testing in Unity (https://github.com/microsoft/AirSim/pull/2624#issuecomment-627699236)
- [x] Documentation
- [x] Python examples replacement
- [x] Replacements in ROS wrapper

### CODE REVIEW

File: ros/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp [hunk 1]
```diff
@@ -804,7 +809,9 @@ void AirsimROSWrapper::drone_state_timer_cb(const ros::TimerEvent& event)
         if (has_gimbal_cmd_)
         {
             std::unique_lock<std::recursive_mutex> lck(drone_control_mutex_);
-            airsim_client_.simSetCameraOrientation(gimbal_cmd_.camera_name, gimbal_cmd_.target_quat, gimbal_cmd_.vehicle_name);
+            // Only camera rotation, no translation movement of camera
+            airsim_client_.simSetCameraPose(gimbal_cmd_.camera_name, get_airlib_pose(0, 0, 0, gimbal_cmd_.target_quat),
+                                            gimbal_cmd_.vehicle_name);
```
Comment (rajat2004) [inline ros/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp:814]: This change needs a review, in regards to whether new fields should be added to the Gimbal message, current implementation, etc.

**Comment by rajat2004** [comment]
Might be better to merge this after #2712 just to make sure

**Comment by mihailk239** [comment]
Please add same functional for python client.

**Comment by rajat2004** [comment]
@mihailk239 The Python API change is there in the PR, see `client.py`
Should there be a backwards compatible layer, which calls the new API with 0 position? @madratman @saihv

File: ros/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp (hunk 1 above)
Comment (madratman) [reply ros/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp]: I think it's okay, as Gimbal usually implies an orientation only.

**Pull request merged**